src/sentry/integrations/slack/link_team.py

```diff
@@ -0,0 +1,203 @@
+from django import forms
+from django.http import Http404
+from django.urls import reverse
+from django.views.decorators.cache import never_cache
+from rest_framework.response import Response
+
+from sentry.models import (
+    ExternalActor,
+    Identity,
+    IdentityProvider,
+    Integration,
+    NotificationSetting,
+    OrganizationMember,
+    Team,
+)
+from sentry.notifications.types import NotificationSettingOptionValues, NotificationSettingTypes
+from sentry.shared_integrations.exceptions import ApiError
+from sentry.types.integrations import ExternalProviders
+from sentry.utils.http import absolute_uri
+from sentry.utils.signing import sign, unsign
+from sentry.web.decorators import transaction_start
+from sentry.web.frontend.base import BaseView
+from sentry.web.helpers import render_to_response
+
+from .client import SlackClient
+from .utils import logger
+
+
+def build_linking_url(integration, slack_id, channel_id, channel_name, response_url):
+    signed_params = sign(
+        integration_id=integration.id,
+        slack_id=slack_id,
+        channel_id=channel_id,
+        channel_name=channel_name,
+        response_url=response_url,
+    )
+
+    return absolute_uri(
+        reverse("sentry-integration-slack-link-team", kwargs={"signed_params": signed_params})
+    )
+
+
+class SelectTeamForm(forms.Form):
+    team = forms.ChoiceField(label="Team")
+
+    def __init__(self, teams, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields["team"].choices = [(team.id, team.slug) for team in teams]
+        self.fields["team"].widget.choices = self.fields["team"].choices
+
+
+class SlackLinkTeamView(BaseView):
+    def get_identity(self, integration, slack_id):
+        try:
+            idp = IdentityProvider.objects.get(type="slack", external_id=integration.external_id)
+        except IdentityProvider.DoesNotExist:
+            logger.error(
+                "slack.action.invalid-team-id", extra={"slack_id": integration.external_id}
+            )
+            return self.respond(status=403)
```

Comment:
`BaseView.respond()` requires a template. We can create and `error.html` page that just displays a message or re-render the same page but with an error toast or banner.

Reply:
good catch, updated